### PR TITLE
Define the __eq__ method of DBusData

### PR DESCRIPTION
**Pull request by poncovka**
Compare the values of all data fields by default.

Use a dictionary from `generate_dictionary_from_data`
to generate a string representation of the data. Change the
argument `add` to a dictionary of attributes to make it more
user-friendly.

### CODE REVIEW

File: pyanaconda/dbus/structure.py [hunk 1]
```diff
@@ -273,23 +273,21 @@ def generate_string_from_data(obj, skip=None, add=None):
...
-        attributes.append(attribute)
+    for name in add or dict():
+        dictionary[name] = add[name]
 
-    if extra_attributes:
-        attributes.extend(extra_attributes)
-        attributes.sort()
+    attributes = sorted([
+        "{}={}".format(name, repr(value))
+        for name, value in dictionary.items()
```
Comment (jkonecny12) [inline pyanaconda/dbus/structure.py]: Can you please merge this to one line?

File: pyanaconda/dbus/structure.py [hunk 2]
```diff
@@ -273,23 +273,21 @@ def generate_string_from_data(obj, skip=None, add=None):
...
 
-    for field in get_fields(obj).values():
-        if field.data_name in skipped_names:
-            continue
+    for name in skip or list():
+        dictionary.pop(name, None)
 
-        attribute = "{}={}".format(field.data_name, repr(field.get_data(obj)))
-        attributes.append(attribute)
+    for name in add or dict():
```
Comment (jkonecny12) [inline pyanaconda/dbus/structure.py]: I don't like this construct but it works so if you want, you can leave it this way.

File: pyanaconda/dbus/structure.py (hunk 1 above)
Comment (poncovka) [reply pyanaconda/dbus/structure.py]: Fixed.

**Comment by poncovka** [comment]
Updated.

**Pull request merged**